embedded/store/ongoing_tx.go

```diff
@@ -323,7 +339,7 @@ func (tx *OngoingTx) GetWithPrefixAndFilters(prefix, neq []byte, filters ...Filt
 	}
 
 	key, valRef, err = tx.snap.GetWithPrefixAndFilters(prefix, neq, filters...)
-	if errors.Is(err, ErrKeyNotFound) {
+	if !tx.readOnly && errors.Is(err, ErrKeyNotFound) {
```

Comment:
minor comment, but if we can combine this and line 360 under the same condition for tx.readOnly, would be very readable